community/cypher/cypher-compiler-2.2/src/main/scala/org/neo4j/cypher/internal/compiler/v2_2/ast/Pattern.scala

```diff
@@ -274,12 +274,27 @@ case class RelationshipPattern(
     }
 
   def semanticCheck(ctx: SemanticContext): SemanticCheck =
+    checkNoInvalidSelfReferences chain
     checkNoOptionalRelsForAnExpression(ctx) chain
-    checkNoVarLengthWhenUpdating(ctx) chain
-    checkNoLegacyOptionals(ctx) chain
-    checkNoParamMapsWhenMatching(ctx) chain
-    checkProperties(ctx) chain
-    checkNotUndirectedWhenCreating(ctx)
+      checkNoVarLengthWhenUpdating(ctx) chain
+      checkNoLegacyOptionals(ctx) chain
+      checkNoParamMapsWhenMatching(ctx) chain
+      checkProperties(ctx) chain
+      checkNotUndirectedWhenCreating(ctx)
+
+
+  private def checkNoInvalidSelfReferences: SemanticCheck = {
+    when(!isSingleLength) {
+      val (hasSelfReference, pos) = properties.treeFold((false, position)) {
```

Comment:
Indeed, the included test already generates an error: "Type mismatch: expected Map, Node or Relationship but was Collection<Relationship>] (line 1, column 25 (offset: 24))"